src/components/TodoItem/TodoItem.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import "./TodoItem.scss";
+import { useState, useContext } from "react";
+import { apiBaseUrl, requestDefaultHeaders } from "../../../config";
+import { StatusLengthContext } from "../../../StatusLengthContext";
+import { formatDate } from "../../utils";
+
+const TodoItem = ({ text, createdAt, taskId, getTasks }) => {
+  const [error, setError] = useState("");
+  const { todosCount, setTodosCount } = useContext(StatusLengthContext);
+
+  const deleteTodo = () => {
+    try {
+      setError("");
+      fetch(`${apiBaseUrl}/todos/${taskId}`, {
+        method: "DELETE",
+        headers: requestDefaultHeaders,
+      });
+      getTasks();
+      setTodosCount(todosCount.all - 1);
```

Comment:
1. State is an object but you are setting it as number.
2. When you delete a todo, based on what is todos's status you should change the correct property.